examples/classic_controllers/classic_controllers.py

```diff
@@ -704,41 +902,507 @@ def control(self, state, reference):
+        psi_max = min(psi_opt, psi_max_)
+
+        psi_max_idx = self.get_psi_idx_mtpf(psi_max)
+        t_max = np.abs(self.mtpf[psi_max_idx, 1])
+        if np.abs(torque) > t_max:
+            torque = np.sign(torque) * t_max
+
+        if self.torque_control == 'online':
+            i_d, i_q = self.get_i_d_q(torque, psi_max, psi_idx_)
+        else:
+            t_idx = self.get_t_idx(torque)
+            psi_idx = self.get_psi_idx(psi_max)
+
+            if self.i_d_inter[t_idx, psi_idx] <= self.mtpf[psi_max_idx, 2]:
+                i_d = self.mtpf[psi_max_idx, 2]
+                i_q = np.sign(torque) * np.abs(self.mtpf[psi_max_idx, 3])
+                torque = np.sign(torque) * t_max
+            else:
+                i_d = self.i_d_inter[t_idx, psi_idx]
+                i_q = self.i_q_inter[t_idx, psi_idx]
+                if i_d > self.mtpc[psi_idx_, 1]:
+                    i_d = self.mtpc[psi_idx_, 1]
+                    i_q = np.sign(torque) * np.abs(self.mtpc[psi_idx_, 2])
+        if i_d < self.mtpf[psi_max_idx, 2]:
+            i_d = self.mtpf[psi_max_idx, 2]
+            i_q = np.sign(torque) * np.abs(self.mtpf[psi_max_idx, 3])
+
+        if self.plot_torque:
+            if self.k == 0:
+                self.intitialize_torque_plot()
+
+            self.k_list.append(self.k / 10000)
+            self.i_d_list.append(i_d)
+            self.i_q_list.append(i_q)
+            self.torque_list.append(torque)
+            self.psi_list.append(psi_max)
+
+            if self.k % self.update_interval == 0:
+                self.psi_plot.scatter(self.torque_list, self.psi_list, c='tab:blue', s=3)
+                self.torque_plot.scatter(self.psi_list, self.torque_list, c='tab:blue', s=3)
+                self.i_d_q_characteristic_.scatter(self.i_d_list, self.i_q_list, c='tab:blue', s=3)
+
+                self.fig_torque.canvas.draw()
+                self.fig_torque.canvas.flush_events()
+                self.k_list = []
+                self.i_d_list = []
+                self.i_q_list = []
+                self.torque_list = []
+                self.psi_list = []
+
+        i_q = np.clip(i_q, -self.nominal_values[self.i_sq_idx], self.nominal_values[self.i_sq_idx]) / self.limit[self.i_sq_idx]
+        i_d = np.clip(i_d, -self.nominal_values[self.i_sd_idx], self.nominal_values[self.i_sd_idx]) / self.limit[self.i_sd_idx]
+
+        self.k += 1
+
+        return i_q, i_d
+
+    def modulation_control(self, state):
+        eps = state[self.epsilon_idx] * self.limit[self.epsilon_idx]
+        u_sd = state[self.u_sd_idx] * self.limit[self.u_sd_idx]
+        u_sq = state[self.u_sq_idx] * self.limit[self.u_sq_idx]
+        u_alpha, u_beta = self.q([u_sd, u_sq], eps)
+        a = 2 * np.sqrt(u_alpha ** 2 + u_beta ** 2) / self.u_dc
+
+        if a > 1.1 * self.a_max:
+            self.integrated = self.integrated_reset
+
+        a_delta = self.k_ * self.a_max - a
+        if state[self.omega_idx] > 0.0001:
```

Comment:
In my opinion, it should be `np.abs(state[self.omega_idx])`. Currently, it enters the else clause for every negative speed value. Or is this intended? Furthermore, I think you can reduce the if-else clause completely by just writing sth. like the following:
```
# Avoid division by zero at 0 speed
omega = max(np.abs(state[self.omega_idx]), 0.001)
psi_max_ = self.u_dc / (np.sqrt(3) * self.limit[self.omega_idx] * omega * self.p)
```